### PR TITLE
feat: Login page

### LINKED ISSUE #4: Auth: App (Login)
Depends: HF5-Project/api#9 HF5-Project/overnites#5 HF5-PROJECT/api#35

Create UI and call to the Auth Endpoint for login

### PR DESCRIPTION
**Pull request by frederikpyt**
## Description
1. Added login page
2. Added user page
3. Added login/logout flow

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -37,6 +37,8 @@
 /packages/flutter/coverage/
 version
 
+# Cookies
+.cookies
```
Comment (kristianbinau) [inline .gitignore:41]: Has the cookies/refresh token things been tested on a real phone? Where do we save the cookies?

File: lib/app/factories/bearer_token_factory.dart [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import 'package:overnites/app/factories/box_factory.dart';
+import 'package:overnites/app/services/bearer_token.dart';
+
+class BearerTokenFactory {
+  static Future<TokenService> make() async {
+    return TokenService(await BoxFactory.make('myBox'));
```
Comment (kristianbinau) [inline lib/app/factories/bearer_token_factory.dart]: Should we give it another name than myBox? Maybe just `overnitesStorage`?

File: lib/app/services/auth.dart [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+import 'package:dio/dio.dart';
+import 'package:overnites/app/services/bearer_token.dart';
+
+class AuthService {
+  AuthService(this._dio, this._bearerTokenService);
+
+  final Dio _dio;
+  final TokenService _bearerTokenService;
+
+  String decodeBase64(String str) {
```
Comment (kristianbinau) [inline lib/app/services/auth.dart]: Private and static?

File: lib/app/services/auth.dart [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        output += '=';
+        break;
+      default:
+        throw Exception('Illegal base64url string!"');
+    }
+
+    return utf8.decode(base64Url.decode(output));
+  }
+
+  Future<bool> verifyToken(String accessToken) async {
```
Comment (kristianbinau) [inline lib/app/services/auth.dart]: Private and static?

File: lib/app/services/auth.dart [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+      final accessToken = _bearerTokenService.getAccessToken();
+
+      if (!(await verifyToken(accessToken))) {
+        await _bearerTokenService.deleteAccessToken();
+
+        if (!(await refresh())) {
+          return false;
+        }
+
+        return isAuthenticated();
```
Comment (kristianbinau) [inline lib/app/services/auth.dart]: Maybe just remove the extra call to isAuthenticated.

```suggestion
        return await refresh();
```

File: lib/app/services/auth.dart [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  Future<bool> logout() async {
+    try {
+      await _dio.post<Map<String, dynamic>>('/auth/logout', data: {});
```
Comment (kristianbinau) [inline lib/app/services/auth.dart:56]: Do we have an issue to make this?

File: lib/app/services/auth.dart (hunk 4 above)
Comment (kristianbinau) [inline lib/app/services/auth.dart]: Maybe this could even be a function of bearer_token_factory instead

File: test/app/view/app_test.dart [hunk 7]
```diff
@@ -1,12 +1,12 @@
...
 void main() {
   group('App', () {
-    testWidgets('renders CounterPage', (tester) async {
+    testWidgets('renders HotelPage', (tester) async {
       await tester.pumpWidget(const App());
-      expect(find.byType(CounterPage), findsOneWidget);
+      expect(find.byType(HotelPage), findsOneWidget);
     });
   });
 }
```
Comment (kristianbinau) [inline test/app/view/app_test.dart:12]: Awesome tests!

**Review by kristianbinau — CHANGES_REQUESTED** [review]
Very good, i have only reviewed the code. I haven't tried it on a device yet.

File: lib/app/services/auth.dart (hunk 6 above)
Comment (frederikpyt) [reply lib/app/services/auth.dart:56]: https://github.com/HF5-PROJECT/api/issues/35

File: lib/app/services/auth.dart (hunk 5 above)
Comment (frederikpyt) [reply lib/app/services/auth.dart]: But then we do not verify the newly fetched token?

File: .gitignore (hunk 1 above)
Comment (frederikpyt) [reply .gitignore:41]: 1. It has not been tested
2. It saves the refresh cookie on the device's file storage. I do not know where

File: lib/app/services/auth.dart (hunk 4 above)
Comment (frederikpyt) [reply lib/app/services/auth.dart]: I will be moving this to the bearer token factory

File: lib/app/services/auth.dart (hunk 5 above)
Comment (frederikpyt) [reply lib/app/services/auth.dart]: 👍🏻

File: .gitignore (hunk 1 above)
Comment (frederikpyt) [reply .gitignore:41]: Moved to Hive storage 😃

**Review by kristianbinau — APPROVED** [review]
YES YES YES!

**Pull request merged**